pkg/target/target.go

```diff
@@ -244,22 +244,34 @@ func (h *K8sValidationTarget) HandleViolation(result *types.Result) error {
 }
 
 func (h *K8sValidationTarget) MatchSchema() apiextensions.JSONSchemaProps {
-	stringList := &apiextensions.JSONSchemaPropsOrArray{
-		// nullable also allows for the empty string
-		Schema: &apiextensions.JSONSchemaProps{Type: "string", Nullable: true}}
+	// Define some repeatedly used sections
+	stringList := apiextensions.JSONSchemaProps{
+		Type: "array",
+		Items: &apiextensions.JSONSchemaPropsOrArray{
+			Schema: &apiextensions.JSONSchemaProps{Type: "string"},
+		},
+	}
+	nullableStringList := apiextensions.JSONSchemaProps{
+		Type: "array",
+		Items: &apiextensions.JSONSchemaPropsOrArray{
+			Schema: &apiextensions.JSONSchemaProps{Type: "string", Nullable: true},
+		},
+	}
 	labelSelectorSchema := apiextensions.JSONSchemaProps{
+		Type: "object",
 		Properties: map[string]apiextensions.JSONSchemaProps{
-			// Map schema validation will only work in kubernetes versions > 1.10. See https://github.com/kubernetes/kubernetes/pull/62333
-			//"matchLabels": apiextensions.JSONSchemaProps{
-			//	AdditionalProperties: &apiextensions.JSONSchemaPropsOrBool{
-			//		Allows: true,
-			//		Schema: &apiextensions.JSONSchemaProps{Type: "string"},
-			//	},
-			//},
+			"matchLabels": {
+				Type: "object",
+				AdditionalProperties: &apiextensions.JSONSchemaPropsOrBool{
```

Comment:
Do we need to add x-preserve-unknown-fields to make `AdditionalProperties` structural?

Reply:
Done.